### PR TITLE
[GLUTEN-2886][CH] Set default join spill settings and improve sort/groupby spill settings

### PR DESCRIPTION
**Pull request by taiyang-li**
## What changes were proposed in this pull request?

(Please fill in changes proposed in this fix)

(Fixes: \#2886)

### CODE REVIEW

**Comment by taiyang-li** [comment]
@baibaichen @lgbo-ustc pls help review this pr, thanks!

**Review by liuneng1994 — DISMISSED** [review]
LGTM

File: backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHTransformerApi.scala [hunk 1]
```diff
@@ -122,20 +122,35 @@ class CHTransformerApi extends TransformerApi with Logging {
...
+        // Only set default max_bytes_before_external_group_by for CH when it is not set explicitly.
+        val groupBySpillKey = settingPrefix + "max_bytes_before_external_group_by";
+        if (!nativeConfMap.containsKey(groupBySpillKey)) {
+          val groupBySpillValue = offHeapSize * 0.5
+          nativeConfMap.put(groupBySpillKey, groupBySpillValue.toLong.toString)
+        }
+
+        val joinAlgorithmKey = settingPrefix + "join_algorithm";
+        if (!nativeConfMap.containsKey(joinAlgorithmKey)) {
+          nativeConfMap.put(joinAlgorithmKey, "grace_hash")
```
Comment (baibaichen) [inline backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHTransformerApi.scala:141]: we need review perforamnce

Comment (taiyang-li) [reply backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHTransformerApi.scala:141]: It is a default settings to make sure OOM do not happen while executing join. Spark users could override default settings if they want to try other join algorithms.

Comment (taiyang-li) [reply backends-clickhouse/src/main/scala/io/glutenproject/backendsapi/clickhouse/CHTransformerApi.scala:141]: default settings of join_algorithm = grace_join already removed.

**Review by liuneng1994 — APPROVED** [review]
LGTM

**Pull request merged**